lib/models/authentication/authentication_service.dart

```diff
@@ -16,6 +18,24 @@ class AuthenticationService {
   })  : _googleSignIn = googleSignIn ?? GoogleSignIn(),
         _firebaseAuth = firebaseAuth ?? FirebaseAuth.instance,
         _facebookAuth = facebookAuth ?? FacebookAuth.instance;
+
+  Future<UserDetail> signUp({required SignUpType signUpType}) async {
+    UserDetail userDetail;
+
+    // CHECK authe service type
+    switch (signUpType) {
+      case SignUpType.googleService:
+        userDetail = await signUpWithGoogle();
+        break;
+      case SignUpType.facebookService:
+        userDetail = await signUpWithFacebook();
+        break;
+      default:
+        throw (UserCancelException());
```

Comment:
this should throw `UnimplementedError` with message to tell developer about this error, because this is developer's fault, not user's fault